contracts/strategies/NFTVoteWinnerTakeAll.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+import { ERC721 } from "@solady/tokens/ERC721.sol";
+import { ERC20 } from "@solady/tokens/ERC20.sol";
+import { BaseStrategy } from "./BaseStrategy.sol";
+import { Transfer } from "../core/libraries/Transfer.sol";
+
+contract NFTVoteWinnerTakeAll is BaseStrategy {
+    uint startTime;
+    uint endTime;
+    ERC721 nft;
+    address winner;
+
+    address[] memory recipients;
+    mapping(address => bool) public isRecipient;
+    mapping(uint => bool) public isVoted;
+    mapping(address => uint) public votes;
+
+    function initialize(bytes32 _identityId, uint256 _poolId, bytes memory _data) external override {
+        super.initialize(_identityId, _poolId, _data);
+        (nft, startTime, endTime) = ERC721(abi.decode(_data, (address, uint, uint)));
+
+        require(startTime < endTime, "start time must be before end time");
+        require(endTime > block.timestamp, "end time must be in the future");
+    }
+
+    function registerRecipients(bytes memory _data, address _sender) external payable returns (address) {
+        require(block.timestamp >= startTime, "voting has not started");
+        require(block.timestamp <= endTime, "voting has ended");
+
+        address[] memory newRecipients = abi.decode(_data, (address[]));
+
+        // check to ensure gas for determining winner doesn't exceed block gas limit (just made up number for now)
+        require(recipients.length + newRecipients.length < 1000, "max 1000 recipients");
```

Comment:
receipients count restriction can be removed 
[see the following comment](https://github.com/allo-protocol/allo-v2/pull/58/files#r1263554934)